src/opentracing/tracer.js

```diff
@@ -79,7 +79,7 @@ class DatadogTracer extends Tracer {
       startTime: fields.startTime
     })
 
-    span.context()._hostname = this._hostname
+    span._spanContext._hostname = this._hostname
```

Comment:
I would prefer to avoid accessing private properties whenever possible. Maybe the `Span` constructor could accept properties for initializing its span context?